pipeline/frontend/yaml/compiler/convert.go

```diff
@@ -55,6 +55,18 @@ func (c *Compiler) createProcess(name string, container *yaml_types.Container, s
 		})
 	}
 
+	var extraHosts []backend_types.HostAlias
```

Comment:
As the resulting array length is known it is better to initialize it either with enough capacity or with needed length initialy:
extraHosts := make([]..., len(container.ExtraHosts))

And the assign values already by loop index